pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -340,6 +341,7 @@ func (gb *GraphBuilder) Run(stopCh <-chan struct{}) {
 		if monitor.stopCh != nil {
 			stopped++
 			close(monitor.stopCh)
+			monitor.stopCh = nil
```

Comment:
I think either this or setting `gb.monitors` to empty at the end of this function will fix the race?

Reply:
This modification also have data race, because monitor.stopCh is also used by 
```
func (m *monitor) Run() {
	m.controller.Run(m.stopCh)
}
``` 
I think your suggestion may be better.